src/main/java/com/ibm/cloud/cloudant/kafka/transforms/predicates/IsDesignDocument.java

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright © 2022 IBM Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file
+ * except in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the
+ * License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied. See the License for the specific language governing permissions
+ * and limitations under the License.
+ */
+package com.ibm.cloud.cloudant.kafka.transforms.predicates;
+
+import java.util.Map;
+import org.apache.kafka.common.config.ConfigDef;
+import org.apache.kafka.connect.data.Schema;
+import org.apache.kafka.connect.data.Schema.Type;
+import org.apache.kafka.connect.errors.DataException;
+import org.apache.kafka.connect.source.SourceRecord;
+import org.apache.kafka.connect.transforms.predicates.Predicate;
+import org.apache.kafka.connect.transforms.util.Requirements;
+import com.ibm.cloud.cloudant.kafka.common.MessageKey;
+import com.ibm.cloud.cloudant.kafka.common.utils.ResourceBundleUtil;
+import static com.ibm.cloud.cloudant.kafka.common.CloudantConst.CLOUDANT_DESIGN_PREFIX;
+import static com.ibm.cloud.cloudant.kafka.common.CloudantConst.CLOUDANT_DOC_ID;
+
+public class IsDesignDocument implements Predicate<SourceRecord> {
+
+    private static final ConfigDef EMPTY_CONFIG_DEF = new ConfigDef();
+
+    @Override
+    public void configure(Map<String, ?> configs) {
+        // No-op; no configuration
+    }
+
+    @Override
+    public ConfigDef config() {
+        return EMPTY_CONFIG_DEF;
+    }
+
+    @Override
+    public boolean test(SourceRecord record) {
+        Schema schema = record.keySchema();
+        Object recordKey = record.key();
```

Comment:
The `_id` will also be on the record value, depending on how it is mapped. Are there any pros or cons of that approach instead?

I suppose it all depends on what other transforms the user is likely to make, eg if they have some custom key transforms

Reply:
>Are there any pros or cons of that approach instead?

The reason I used key was because of tombstones (which could equally apply to design docs). We can't read the `_id` from the [`null`] value on a tombstone event. My thinking was that if we'd filtered all the previous events related to that design doc ID it didn't make sense to allow the tombstone to pass.

Beyond that they are much for muchness, if schemas are disabled we still have to assume that the record is an unaltered `Map` of the type we expect.

>I suppose it all depends on what other transforms the user is likely to make, eg if they have some custom key transforms

or indeed some custom value transforms 😄 

Maybe we ust improve the doc to say "filter the design docs before you do any other transformation on the key", wdyt?

Reply:
Added caveat to doc in 33ada0360bce305e20a775a50ccce721a7af3213